src/utils/storeManager.ts

```diff
@@ -9,12 +9,12 @@ export type Store<T> = {
  * A custom store creation utility
  */
 export function createStore<T extends object>(initialState: T): Store<T> {
-  const state = { ...initialState };
+  let state = { ...initialState };
   const listeners = new Set<() => void>();
 
   const setState = (partial: Partial<T> | ((state: T) => Partial<T>)) => {
     const nextState = typeof partial === 'function' ? partial(state) : partial;
-    Object.assign(state, nextState);
+    state = { ...state, ...nextState };
```

Comment:
When updating the state within `setState`, I used the spread syntax (`state = { ...state, ...nextState }`) instead of `Object.assign`. This approach ensures that the state object always has a new reference, allowing `useSyncExternalStore` to recognize that the state has changed.